src/erlang_ls.erl

```diff
@@ -22,38 +23,100 @@ main(Args) ->
   lager:info("Started erlang_ls server", []),
   receive _ -> ok end.
 
+-spec print_version() -> ok.
+print_version() ->
+  {ok, Vsn} = application:get_key(?APP, vsn),
+  io:format("Version: ~s~n", [Vsn]),
+  ok.
+
 %%==============================================================================
 %% Argument parsing
 %%==============================================================================
 
 -spec parse_args([string()]) -> ok.
-parse_args([]) ->
-  ok;
-parse_args(["--transport", Name | Rest]) ->
+parse_args(Args) ->
+  case getopt:parse(opt_spec_list(), Args) of
+    {ok, {[version | _], _BadArgs}} ->
+      print_version(),
+      halt(1);
+    {ok, {ParsedArgs, _BadArgs}} ->
+      %% Handle backwards compatibility of clients that do not provide args but
+      %% only supply an integer. Assume it to be the port and remove the default
+      %% that we pick for the port.
+      ValidArgs = case lists:keyfind(port_old, 1, ParsedArgs) of
```

Comment:
We can address this separately, but do we really need to keep backward compatibility for this right now? @jfacorro

Reply:
This was to support previous versions of the `lsp-mode` client. If we want to force people to update to the latest, then we can remove this support.